app/src/main/java/net/gini/android/gvlexample/GVLExamplePresenter.java

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * Created by Alpar Szotyori on 27.11.2017.
+ *
+ * Copyright (c) 2017 Gini GmbH.
+ */
+
+class GVLExamplePresenter extends GVLExampleContract.Presenter {
+
+    private boolean mGVLLaunchedForImportedFile = false;
+
+    GVLExamplePresenter(final GVLExampleContract.View view) {
+        super(view);
+    }
+
+    @Override
+    void presentInfos() {
+        getView().showInfos();
+    }
+
+    @Override
+    void launchGVL() {
+        getView().requestCameraPermission(new GVLExampleContract.View.PermissionRequestListener() {
+            @Override
+            public void permissionGranted() {
+                doLaunchGVL();
+            }
+
+            @Override
+            public void permissionDenied() {
+            }
+        });
+    }
+
+    private void doLaunchGVL() {
+        // NOTE: on Android 6.0 and later the camera permission is required before checking the requirements
+        RequirementsReport report = GiniVisionRequirements.checkRequirements(getView().getContext());
+        if (!report.isFulfilled()) {
+            getView().showUnfulfilledRequirements(report);
+            return;
+        }
+
+        final Context context = getView().getContext();
+
+        Intent intent = new Intent(context, CameraActivity.class);
+
+        final GiniVisionFeatureConfiguration giniVisionFeatureConfiguration =
+                GiniVisionFeatureConfiguration.buildNewConfiguration()
+                                              .setDocumentImportEnabledFileTypes(
+                                                      DocumentImportEnabledFileTypes.PDF_AND_IMAGES)
+                                              .setFileImportEnabled(
+                                                      true)
+                                              .build();
+
+        intent.putExtra(CameraActivity.EXTRA_IN_GINI_VISION_FEATURE_CONFIGURATION,
+                giniVisionFeatureConfiguration);
+
+        CameraActivity.setReviewActivityExtra(intent, context, ReviewActivity.class);
+        CameraActivity.setAnalysisActivityExtra(intent, context, AnalysisActivity.class);
+
+        getView().showGVL(intent);
+    }
+
+    @Override
+    void launchGVLForImportedFile(final Intent intent) {
+        getView().requestStoragePermission(new GVLExampleContract.View.PermissionRequestListener() {
+            @Override
+            public void permissionGranted() {
+                doLaunchGVLForImportedFile(intent);
+            }
+
+            @Override
+            public void permissionDenied() {
+                getView().finish();
+            }
+        });
+    }
+
+    private void doLaunchGVLForImportedFile(final Intent intent) {
+        try {
+            final Intent giniVisionIntent =
+                    GiniVisionFileImport.createIntentForImportedFile(intent, getView().getContext(),
+                            ReviewActivity.class, AnalysisActivity.class);
+            getView().showGVL(giniVisionIntent);
```

Comment:
It could be the case that the view is not longer available after requesting the permissions, right?

Reply:
I don't think that can happen since the presenter has a strong reference to the view...